pathauto.install

```diff
@@ -51,3 +52,101 @@ function pathauto_update_8001() {
     }
   }
 }
+
+/**
+ * Converts patterns from configuration objects to configuration entities.
+ */
+function pathauto_update_8100(&$sandbox) {
+  $messages = array();
+  $delete_config = TRUE;
+
+  if (!\Drupal::service('module_handler')->moduleExists('ctools')) {
+    throw new UpdateException('Please, install Chaos tools suite (https://www.drupal.org/project/ctools) before running this databsae update.');
+  }
+
+  $entity_manager = \Drupal::service('entity.manager');
+  $entity_info = $entity_manager->getDefinitions();
+
+  // 1. Load all patterns.
+  $config = \Drupal::service('config.factory')->getEditable('pathauto.pattern');
+  $patterns = $config->get('patterns');
+
+  // 2. Create a configuration entity per pattern.
+  foreach ($patterns as $entity_type => $pattern_config) {
+    if (!array_key_exists($entity_type, $entity_info)) {
+      // We found an entity type which we don't know how to process. Report it.
+      $messages[] = t('Entity of type @type was not processed.', array('@type' => $entity_type));
```

Comment:
I'm not sure what to do about forum, to be honest. I'm OK with leaving that out, as far as I know, nobody actually tried to use it in 8.x, it has no tests, so it is very likely broken anyway.